### PR TITLE
cppclient - check for more null cases and return optional in getNodeInfo

### PR DESCRIPTION
**Pull request by tsvisabo**
Return optional on getNodeInfo
treat other null cases

### CODE REVIEW

File: cppclient/api_json.cc [hunk 1]
```diff
@@ -30,8 +30,12 @@ DEFINE_uint32(maxNumAddressesForGetBalances, 1000,
 namespace cppclient {
 
 bool IotaJsonAPI::isNodeSolid() {
-  auto ni = getNodeInfo();
-
+  auto niOptional = getNodeInfo();
```
Comment (th0br0) [inline cppclient/api_json.cc]: let's call these `maybeNodeInfo`?

File: cppclient/api_json.cc [hunk 2]
```diff
@@ -30,8 +30,12 @@ DEFINE_uint32(maxNumAddressesForGetBalances, 1000,
...
+  if (!niOptional.has_value()) {
+    LOG(INFO) << __FUNCTION__ << " request failed.";
+    return {};
```
Comment (th0br0) [inline cppclient/api_json.cc]: this should return `false` explicitly

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: cppclient/api_json.cc (hunk 2 above)
Comment (tsvisabo) [reply cppclient/api_json.cc]: definitley

File: cppclient/api_json.cc (hunk 1 above)
Comment (tsvisabo) [inline cppclient/api_json.cc]: maybe :)

**Review by th0br0 — APPROVED** [review]

**Comment by th0br0** [comment]
Rebase on master please

**Pull request merged**